qhana_plugin_registry/tasks/plugin_filter.py

```diff
@@ -73,7 +76,24 @@ def get_plugins_from_filter(filter_dict: dict, plugin_mapping: dict) -> set[RAMP
                 if Version(p.version) in specifier
             }
         case {"name": name}:
-            return {p_id for p_id, p in plugin_mapping.items() if name == p.name}
+            plugin_ids = set()
+            for p_id, p in plugin_mapping.items():
+                sm = SequenceMatcher(None, name.lower(), p.name.lower())
+                if sm.ratio() > 0.8:
+                    plugin_ids.add(p_id)
+            return plugin_ids
+        case {"id": plugin_id}:
+            # match plugin id or plugin id without version
+            plugin_ids = set()
+            for p_id, p in plugin_mapping.items():
+                if plugin_id == p.plugin_id:
+                    plugin_ids.add(p_id)
+                elif (
+                    "@" in p.plugin_id
+                    and plugin_id == p.plugin_id[: p.plugin_id.rfind("@")]
+                ):
+                    plugin_ids.add(p_id)
```

Comment:
I would reverse the logic a bit:

```python
if '@' not in plugin_id and '@' in p.plugin_id:
    plugin_id += '@'  # <- ensures that the full id matches
    if p.plugin_id.startswith(plugin_id):
        # match
else:
    if p.plugin_id == plugin_id:
        # match
```

This should make this test easier to read.

Or split both ids at the '@' and compare from there?

Reply:
I implemented it using `split()`.

```python
if plugin_id == p.plugin_id:
    # match
elif plugin_id.split("@") == p.plugin_id.split("@")[:-1]:
    # match
```

For cases where ids have multiple `@`s and same prefixes, e.g. `asdf@123@v0`, `asdf@000@v1`, I matched the prefix until the last `@` for plugin id filters without version. Although we probably won't have such plugin ids in practice the hypothesis tests may generate them.